### PR TITLE
Rename tetris_concat

### LINKED ISSUE #5556: Rename tetris_concat?
I like the name, but seems like we might want to change it before 1.0 for trademark or whatever? Maybe rigid_concat or something?

### PR DESCRIPTION
**Pull request by daxfohl**
Fixes #5556

### CODE REVIEW

File: cirq-core/cirq/circuits/circuit.py
Comment (maffoo) [inline cirq-core/cirq/circuits/circuit.py:1456]: Maybe mention "ragged" edges here, e.g.
```suggestion
        """Concatenates circuits, overlapping them if possible due to ragged edges.
```

**Review by maffoo — APPROVED** [review]
Looks good, though I think it's worth getting approval from another maintainer to have more eyes on the name.

**Comment by daxfohl** [comment]
Added bikeshed session to cynq schedule tomorrow. (including, do we even want to do this at all)

**Pull request merged**
